src/types/cql-patient.interfaces.ts

```diff
@@ -1,11 +1,25 @@
-import { Record } from '../cql-patient';
-
+/*
+ * Iterator for the patients provided to the execution engine
+ */
 export interface DataProvider {
-  patients: any[];
   currentPatient(): any;
   nextPatient(): any;
 }
 
+/*
+ * Structure of a Record
+ */
 export interface RecordObject {
-  [recordType: string]: Record[];
+  get(field: any): any;
+  getId(): any;
+  getCode(field: any): any;
+  getDate(field: any): any;
+  getDateOrInterval(field: any): any;
+}
+
+/*
+ * Patient data object that implements logic for searching for records based on the Patient
+ */
+export interface PatientObject {
```

Comment:
I think this might need to extend `RecordObject` as well (or at least some of its functions)...  although I'm not 100% sure.  At least `getId()` -- and _maybe_ some of the others.

Reply:
Yeah this is definitely right, since `Patient` both `extends Record` and `implements PatientObject`. Addressed in [c139c09](https://github.com/cqframework/cql-execution/pull/260/commits/c139c09f7e72b4a6650dfe056132fefd342f24fa)